example/Benchmark.cpp

```diff
@@ -0,0 +1,380 @@
+    std::ofstream file(filename);
+    file << std::setprecision(14);
+    for (const auto& val : vec) {
+        file << val << "\n";
+    }
+}
+
+
+
+// == Main loop ===
+int main(int argc, char* argv[]) {
+    if (argc < 5) {
+        std::cerr << "Usage: " << argv[0] << " path/to/output nRho nT sp5_file matid1 matid2 ..."; //Is this good usage? 
+        return 1;
+    }
+    fs::path base_output_path(argv[1]);
+    //if the folder does not exist, try to make one
+    if (!fs::exists(base_output_path)) {
+        if (!fs::create_directories(base_output_path)) {
+            std::cerr << "Failed to create output directory: " << base_output_path << "\n";
+            return 1; }}
+    int nRho = std::stoi(argv[2]);  //test on batch sizes of 144, 269, 512, 2048 or round powers of ten
+    int nT = std::stoi(argv[3]); 
+    std::string sp5file = argv[4];
+
+    std::vector<int> matids;
+    for (int i = 5; i < argc; ++i)
+        matids.push_back(std::atoi(argv[i]));
+
+    // == Iterate through each material ==
+    for (int matid : matids) {
+        const int ntimes = 20; // Perform 20 time trials
+
+        // == Get material metadata ==
+        SesameMetadata meta;
+        eosGetMetadata(matid, meta);
+
+        // == Set up bounds ==
+        Real rhoMin = 1.1 * std::max(meta.rhoMin, 1e-5);
+        Real rhoMax = 0.9 * meta.rhoMax;
+        Real TMin   = 1.1 * std::max(meta.TMin, 1.0);
+        Real TMax   = 0.9 * meta.TMax;
+        Bounds rhoBounds(rhoMin, rhoMax, nRho);
+        Bounds TBounds(TMin, TMax, nT);
+
+        std::vector<Real> rhos(nRho), temps(nT);
+        for (int i = 0; i < nRho; ++i) rhos[i] = rhoBounds.i2lin(i);
+        for (int j = 0; j < nT; ++j) temps[j] = TBounds.i2lin(j);
+
+        // == This will save the values of rho and T corresponding to the indices chosen ==
+        std::string rho_file = (base_output_path / ("rho_axis_" + std::to_string(matid) + "_nRho-" + std::to_string(nRho) + ".csv")).string();
+        std::string temp_file = (base_output_path / ("temp_axis_" + std::to_string(matid) + "_nT-" + std::to_string(nT) + ".csv")).string();
+        write_vector_csv(rho_file, rhos);
+        write_vector_csv(temp_file, temps);
+
+        // === Load EOS Models ===
+        SpinerEOSDependsRhoT eos_rt(sp5file, matid);
+        SpinerEOSDependsRhoSieTransformable<ShiftTransform> eos_rs(sp5file, matid); //eventually make it SpinerEOSDependsRhoSie<NullTransfom>, for example
+	EOSPAC eos_ref(matid); 
+
+        //These vectors will store the compute time for each model for the 20 trials
+        std::vector<double> time_sie_eos_ref_list, time_sie_rt_list, time_sie_rs_list;
+        std::vector<double> time_P_eos_ref_T_list, time_P_rt_T_list, time_P_rs_T_list;
+        std::vector<double> time_P_eos_ref_sie_list, time_P_rt_sie_list, time_P_rs_sie_list;
+        std::vector<double> time_T_back_eos_ref_list, time_T_back_rt_list, time_T_back_rs_list;
```

Comment:
Minor quibble: if the number of trials is known at compile time, these can have compile-time lengths or they can be `std::array`. But for the sake of a test, I don't think any of this matters, so don't bother changing it.